pkg/build/controller/controller.go

```diff
@@ -317,7 +317,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			}
 		}
 	case kapi.PodFailed:
-		build.Status.Reason = ""
+		// NOTE: why do we need to reset the reason for a failed Pod???
+		// build.Status.Reason = ""
```

Comment:
i think the safe thing to do is check if the Reason is one of your "failed" reasons and if so, then don't reset it to "".  But otherwise you need to, because if the pod enters the failed state for some other reason, you don't want to preserve the Reason from when the pod entered the running state, for example.